homeassistant/components/drop/__init__.py

```diff
@@ -0,0 +1,69 @@
+"""The DROP integration."""
+
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components import mqtt
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    CONF_COORDINATOR,
+    CONF_DATA_TOPIC,
+    CONF_DEVICE_TYPE,
+    CONF_UNIQUE_ID,
+    DOMAIN as DROP_DOMAIN,
+)
+from .coordinator import DROP_DeviceDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up DROP from a config entry."""
+
+    # Make sure MQTT integration is enabled and the client is available.
+    if not await mqtt.async_wait_for_mqtt_client(hass):
+        _LOGGER.error("MQTT integration is not available")
+        return False
+
+    hass.data.setdefault(DROP_DOMAIN, {})
+    hass.data[DROP_DOMAIN][config_entry.entry_id] = {}
+    hass.data[DROP_DOMAIN][config_entry.entry_id][
+        CONF_COORDINATOR
+    ] = DROP_DeviceDataUpdateCoordinator(hass, config_entry)
```

Comment:
I simplified the initialization in 0b5411d4ea64a5bd9f76186ae596dbd5d0e60348 as such:

`hass.data.setdefault(DROP_DOMAIN, {})[config_entry.entry_id] = {}`
`hass.data[DROP_DOMAIN][config_entry.entry_id][CONF_COORDINATOR] = DROP_DeviceDataUpdateCoordinator(hass, config_entry)`

Your suggestion was good, but fails because the `[config_entry.entry_id]` key needs to exist before the `[CONF_COORDINATOR]` key can be created.